src/gretel_trainer/benchmark/comparison.py

```diff
@@ -42,304 +27,37 @@
-
-        self.config.working_dir.mkdir(exist_ok=True)
-        self.datasets = [
-            _standardize_dataset(dataset, self.config.working_dir)
-            for dataset in datasets
-        ]
-        self._gretel_executors: dict[RunKey, Executor] = {}
-        self._custom_executors: dict[RunKey, Executor] = {}
-        self.trainer_project_names: dict[str, str] = {}
-        self.thread_pool = ThreadPoolExecutor(5)
-        self.futures: dict[FutureKeyT, Future] = {}
-
-        self._report_scores = {
-            score_name: ALL_REPORT_SCORES[score_name]
-            for score_name in ["SQS"] + self.config.additional_report_scores
-        }
-
-    @property
-    def executors(self) -> dict[RunKey, Executor]:
-        return self._gretel_executors | self._custom_executors
-
-    def _make_project(self) -> Project:
-        display_name = self.config.project_display_name
-        if self.config.trainer:
-            display_name = f"{display_name}-evaluate"
-
-        return create_project(display_name=display_name)
-
-    def prepare(self) -> Comparison:
-        self._project = self._make_project()
-
-        _trainer_project_index = 0
-        for dataset in self.datasets:
-            # TODO: avoid upload for public datasets, when possible
-            artifact_key = _upload_dataset_to_project(
-                dataset.data_source,
-                self._project,
-                self.config.trainer,
-            )
-
-            for model in self.gretel_models:
-                self._setup_gretel_run(
-                    dataset, model, artifact_key, _trainer_project_index
-                )
-                _trainer_project_index += 1
-
-            for model in self.custom_models:
-                self._setup_custom_run(dataset, model, artifact_key)
-
-        return self
-
-    def execute(self) -> Comparison:
-        for run_key, executor in self._gretel_executors.items():
-            self.futures[run_key] = self.thread_pool.submit(_run_gretel, executor)
-
-        if len(self._custom_executors) > 0:
-            self.futures["custom"] = self.thread_pool.submit(
-                _run_custom, list(self._custom_executors.values())
-            )
-
-        return self
-
-    @property
-    def results(self) -> pd.DataFrame:
-        result_records = [self._result_dict(run_key) for run_key in self.executors]
-        return pd.DataFrame.from_records(result_records)
+) -> Session:
+    config = config or BenchmarkConfig()
 
-    def export_results(self, destination: str) -> None:
-        self.results.to_csv(destination, index=False)
+    model_instances = _create_models(models)
+    _validate_compare(model_instances, datasets)
 
-    def wait(self) -> Comparison:
-        [future.result() for future in self.futures.values()]
-        return self
+    # TODO(pm): I think moving this outside of Session makes sense, so it can be configured externally
+    configure_session(api_key="prompt", cache="yes", validate=True)
```

Comment:
You mean outside of any Benchmark library code entirely, right? Yes, I agree—in fact, we made that change for [Trainer](https://github.com/gretelai/trainer/pull/86) and [Relational](https://github.com/gretelai/trainer/pull/87) already to support running in hybrid environments; I think I just missed Benchmark due to the nature of the long-running `b2` branch.

Reply:
Yes, that was my thinking.
Should I remove it then? Or leave as-is to be handled in a separate PR?

Reply:
You can go ahead and take it out now.